Edit screen functionality added + sync with accordion.

renm10 opened the pull request:
# Description

In this PR, the edit screens has been added functionality including:
- Create an activity 
- Edit an activity
- Delete an activity

Activities are now separated by their own date, meaning in the accordion of April 9th, only the activities that happen in April 9th appear.

**Created/Edited/Deleted events are all synced in the activity view in the accordion.**

Fixes #109 

# Add an image of the change (if applicable)
### Pressing the edit button shows the edit screen
![right-way-edit-button](https://github.com/UNLV-CS472-672/2024-S-GROUP6-RW/assets/43124584/94a6bc86-1b53-4046-b161-8558639194e4)
### Create event button pops a create event dialog
![right-way-create-event-button](https://github.com/UNLV-CS472-672/2024-S-GROUP6-RW/assets/43124584/72652e43-d734-4c08-ab4b-eb5596b30dce)
### Edit button pops a edit event dialog (Invisible delete button on top right corner of dialog)
![edit-event-button](https://github.com/UNLV-CS472-672/2024-S-GROUP6-RW/assets/43124584/f90b0e01-d257-48e9-8fc7-15a18523fee0)
### All changes are synced to the itinerary accordion
![itinerary-accordion-syncview](https://github.com/UNLV-CS472-672/2024-S-GROUP6-RW/assets/43124584/4c61b37c-8897-4b31-9591-23a421d0ac80)

# Type of change

- [ ] Bug fix 
- [x] New feature 
- [ ] Breaking change 
- [ ] This change requires a documentation update

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

Niblick1020 reviewed (changes requested):
The Save button is not working.

![image](https://github.com/UNLV-CS472-672/2024-S-GROUP6-RW/assets/46581757/180212f5-a70f-4d9b-a25d-9ac36e443b8b)

Whenever I click save nothing loads.

renm10 commented:
@Niblick1020 Thanks for letting me know!
It was a problem in the save function where the values of date picker components was only set after changing the values. In the case in the picture, I assume you didn't change the default values so that is why it didn't save. Anyways that is an problem so I fixed it by actually setting the startTime and endTime not to NULL but an actual value.
Also I fixed the merge conflict so it's all ready to be merged!

Niblick1020 reviewed (approved):
I'm going to go ahead and approve this merger. I tried testing it out but because of the current error and master when viewing the itinerary, I cannot test it so we're gonna must roll with it and try it out later. I'll let you know if I run into any hiccups, I'm approving this.
